CONTRIBUTING.md

```diff
@@ -5,6 +5,27 @@ We love your input! We want to make contributing to this project as easy and tra
 
 We use [All Contributors](https://allcontributors.org/docs/en/specification) specification to handle recognitions. For more details read [this](https://github.com/asyncapi/community/blob/master/recognize-contributors.md) document.
 
+## Guidelines for new contributors 
+
+1. Read the [generator docs](https://www.asyncapi.com/docs/tools/generator) to get an overview of what the project is all about.
+2. While reading through the docs one might identify issues (e.g., maybe a link is not working, text formatting issues, etc.).
+3. Some concepts in the documentation might not be presented clearly, so contributors can suggest improvements (e.g., adding more examples or code snippets to make the content more effective).
+4. Familiarize yourself with the project's structure by reviewing the source code and understanding the roles of different files and components.
+5. Testing is one of the areas where new contributors can contribute, as running tests reveals uncovered lines in a particular file (i.e., a **test coverage report**).
+6. You can then try to add tests for those uncovered lines (e.g., a function within a file might not be covered). For reference, you can check out [PR](https://github.com/asyncapi/generator/pull/1379).
+7. Instead of creating entirely new tests, consider enhancing existing ones by adding more edge cases (e.g., if only a few test cases are written, consider including additional scenarios).
```

Comment:
```suggestion
7. Instead of creating entirely new tests, consider enhancing existing ones by adding more edge cases to include additional test scenarios.
```